compaction/compaction_manager.cc

```diff
@@ -1523,8 +1523,7 @@ class rewrite_sstables_compaction_task_executor : public sstables_task_executor
     }
 
     virtual future<sstables::compaction_result> rewrite_sstable(const sstables::shared_sstable sst) {
-        // FIXME: this compaction should run with maintenance priority.
-        co_await coroutine::switch_to(_cm.compaction_sg());
+        co_await coroutine::switch_to(_cm.maintenance_sg());
```

Comment:
Please split this off to its own series, and add an issue. I don't expect we'll backport it (nobody runs rewritesstables), but let's be prepared.

Reply:
Ok